### DISCUSSION TITLE
Netlify functions do not work with `npm run generate`

### DISCUSSION THREAD

**Discussion by dbzx10299** [nuxt/nuxt#24801] [Questions]
I have a site built with nuxt3 and I have it hosted on netlify. I also use netlify forms for a contact form. In my netlify configuration in the netlify ui, I have the build command set to `npm run generate`. This needs to be like this or else my contact form will **NOT** work.

With `npm run generate`, edge functions work, the contact form works, and regular plain netlify serverless functions do **not** work.

If I list `npm run build` as the build command, the edge functions work, the serverless functions work, and the contact form does **not** work.

Edge functions use a deno runtime, and serverless functions use a node runtime, so in some cases, certain npm packages might need to be used in the node runtime and therefore in a regular serverless function.

I would really appreciate if anyone could give me any ideas how this could be resolved. I tried all the netlify nitro presets, that did not help. I also tried using `npm run build --prerender true` and that did not help either.

The interesting this is looking at my netlify deploy, if I run `npm run build` or `npm run generate`, either way I get this message in the deploy log. 

```
2:27:54 AM: Packaging Functions from netlify/functions directory:
2:27:54 AM:  - bar.js
2:27:54 AM: ​
2:27:55 AM: ​
2:27:55 AM: (Functions bundling completed in 698ms)
2:27:55 AM: ​
2:27:55 AM: Edge Functions bundling                                       
2:27:55 AM: ────────────────────────────────────────────────────────────────
2:27:55 AM: ​
2:27:55 AM: Packaging Edge Functions from netlify/edge-functions directory:
2:27:55 AM:  - foo
2:27:55 AM:  - hello
2:27:57 AM: ​
2:27:57 AM: (Edge Functions bundling completed in 1.8s)
```

I am on node version 20, the latest version of nuxt and I've been struggling to get everything working for days and I don't understand why I can't get the serverless functions running because it seems I have to use `npm run generate` for my form to work.

When I run `netlify dev` the serverless functions work, and return json to me, but after `npm run generate` it keeps trying to send back an html document.

**Comment by TheAlexLichter** [comment]
That's true. After `npm run generate` you get only static files.
What you want is `npm run build` with routeRules set so all pages will be prerendered and all links are crawled.

**Reply by dbzx10299** [reply]
When you say all routes prerendered are you referring to writing something like this?

```js
defineNuxtConfig({
  routeRules: {
    '/**': { prerender: true }
  }
})
```

I already tried this code below and the regular serverless functions work now, but the contact form still does not work although it does indeed show that it was prerendered in the `.output` directory. I set the netlify build command to `npm run build` and the contact form is prerendered when looking at the deploy log but netlify does not pick up on it.

```js
defineNuxtConfig({
  routeRules: {
    '/contact': { prerender: true }
  }
})
```
